tests/backends/output/simple.c

```diff
@@ -66,119 +67,180 @@ enum Option_1_code {
   a1 = a2;
-  double b2;
-  option_2 b3;
-  Option_2 b4;
-  char /* bool */ b5;
-  switch (a1.code) {
-    case Bar_No: b5 = 1 /* TRUE */; break;
-    case Bar_Yes: Foo _ = a1.payload.Yes; b5 = 0 /* FALSE */; break;
+  switch (a1->code) {
+    case Bar_No: {
+      b4 = CATALA_TRUE;
+      break;
+    }
+    case Bar_Yes: {
+      b4 = CATALA_FALSE;
+      break;
+    }
   }
-  if (b5) {
-    option_2 b4 = {Option_2_Some_2, {Some_2: 42.}};
-    
+  if (b4 == CATALA_TRUE) {
+    b3->code = Option_2_Some_2;
+    b3->payload.Some_2 = catala_new_dec_str("42");
   } else {
-    option_2 b4 = {Option_2_None_2, {None_2: NULL}};
-    
+    b3->code = Option_2_None_2;
+    b3->payload.None_2 = CATALA_UNITVAL;
   }
-  switch (b4.code) {
-    case Option_2_None_2:
-      double b6;
-      switch (a1.code) {
-        case Bar_No: b6 = 0.; break;
-        case Bar_Yes:
-          Foo foo = a1.payload.Yes;
-          double b6;
-          if (foo.x) {b6 = 1.; } else {b6 = 0.; }
-          b6 = (foo.y + b6);
+  switch (b3->code) {
+    case Option_2_None_2: {
+      CATALA_DEC b5;
+      switch (a1->code) {
+        case Bar_No: {
+          b5 = catala_new_dec_str("0");
           break;
+        }
+        case Bar_Yes: {
+          Foo* foo = a1->payload.Yes;
+          CATALA_DEC b6;
+          if (foo->x == CATALA_TRUE) {
+            b6 = catala_new_dec_str("1");
+          } else {
+            b6 = catala_new_dec_str("0");
+          }
+          b5 = o_add_rat_rat(foo->y, b6);
+          break;
+        }
       }
-      option_2 b3 = {Option_2_Some_2, {Some_2: b6}};
+      b2->code = Option_2_Some_2;
+      b2->payload.Some_2 = b5;
       break;
-    case Option_2_Some_2:
-      double x1 = b4.payload.Some_2;
-      option_2 b3 = {Option_2_Some_2, {Some_2: x1}};
+    }
+    case Option_2_Some_2: {
+      CATALA_DEC x = b3->payload.Some_2;
+      b2->code = Option_2_Some_2;
+      b2->payload.Some_2 = x;
       break;
+    }
   }
-  switch (b3.code) {
-    case Option_2_None_2:
-      catala_raise_fatal_error (catala_no_value,
-        "tests/backends/simple.catala_en", 12, 10, 12, 11);
+  switch (b2->code) {
+    case Option_2_None_2: {
+      static const catala_code_position pos[1] =
+        {{"tests/backends/simple.catala_en", 12, 10, 12, 11}};
+      catala_error(catala_no_value, pos);
+      break;
+    }
+    case Option_2_Some_2: {
+      CATALA_DEC arg = b2->payload.Some_2;
+      b1 = arg;
       break;
-    case Option_2_Some_2: double arg = b3.payload.Some_2; b2 = arg; break;
+    }
   }
-  double b1;
-  b1 = b2;
-  array_1 c2;
-  Array_1 c4;
-  c4.content_field = catala_malloc(sizeof(Array_1));
-  c4.content_field[0] = b1;
-  c4.content_field[1] = b1;
-  option_3 c3 = {Option_3_Some_3, {Some_3: c4}};
-  switch (c3.code) {
-    case Option_3_None_3:
-      catala_raise_fatal_error (catala_no_value,
-        "tests/backends/simple.catala_en", 13, 10, 13, 11);
+  b = b1;
+  c4->size = 2;
+  c4->elements = catala_malloc(2 * sizeof(void*));
```

Comment:
For better readability, intermediate values should have labels like `x_before_check`, `x_after_thing`, etc. Rather than number suffixes.